async.el

```diff
@@ -197,6 +197,25 @@ It is intended to be used as follows:
                              (process-name proc) (process-exit-status proc))))
           (set (make-local-variable 'async-callback-value-set) t))))))
 
+(defun async-read-from-client (proc string)
+  ;; log the message in the process buffer
+  (with-current-buffer (process-buffer proc)
+    (insert string))
+
+  ;; parse message
+  (with-temp-buffer
+    (insert string)
+    (goto-char (point-min))
+    (let (msg)
+      (condition-case _err
+          (while (setq msg (read (current-buffer)))
+            (when-let ((msg-decoded (ignore-errors (base64-decode-string msg))))
+              (setq msg-decoded (car (read-from-string msg-decoded)))
+              (with-current-buffer (process-buffer proc)
+                (when async-callback
+                  (funcall async-callback msg-decoded)))))
+        (error "end of file")))))
```

Comment:
`error` in this case is a handler, not a function. Is this meant to raise an error instead? If not, what's the point of returning the string `"end of file"` from this process filter? BTW, there is also a standard `end-of-file` error symbol, in case that should be caught/raised instead.

Reply:
I'm super sleep deprived, thanks for the review!

Indeed I ment to catch `end-of-file`, this happens when we `read` and reach the end of the chunk the process sent at this moment.

I think errors can just "bubble up" as this code runs in the parent Emacs process.  Although handling errors from process filters is a bit tricky and meh overall :/